views/pages/HomePage.jsx

```diff
@@ -0,0 +1,174 @@
+import React, { useState } from 'react'
+import { useOutletContext } from 'react-router-dom'
+import Select from 'react-select'
+
+import ParticipantsTable from '../components/VirtualTables/ParticipantsTable'
+import { components } from '../forms/ControlledReactSelect/components'
+import { ParticipantsModel } from '../models'
+import { useEffect } from 'react'
+import api from '../api'
+import * as TableHelpers from '../components/VirtualTables/helpers'
+
+const asc = 'ASC'
+const desc = 'DESC'
+const category = 'subject'
+
+const HomePage = () => {
+  const { user, classes, setNotification, setUser } = useOutletContext()
+  const { uid, preferences } = user
+
+  const [participants, setParticipants] = useState([])
+  const [searchResults, setSearchResults] = useState([])
+  const [searchOptions, setSearchOptions] = useState([])
+  const [dimensions, setDimensions] = useState({
+    width: 0,
+    height: 0,
+  })
+  const [direction, setDirection] = useState(asc)
+  const [sortCategory, setSortBy] = useState(category)
+  const [rowCount, setRowcount] = useState(0)
+  const handleResize = () =>
+    setDimensions({ width: window.innerWidth, height: window.innerHeight })
+
+  const fetchParticipants = async () => {
+    const participantsList = await api.participants.all()
+
+    setParticipants(
+      ParticipantsModel.setStarredAndComplete(
+        participantsList,
+        preferences,
+        sortCategory,
+        direction
+      )
+    )
+  }
+
+  const handleUserUpdate = async (e) => {
+    try {
+      const { name, value } = e.target
```

Comment:
What do you think about extracting some of the logic at the beginning of this function to a model we can put some tests around?